src/Avalonia.Controls/HyperlinkButton.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using Avalonia.Controls.Metadata;
+using Avalonia.Threading;
+
+namespace Avalonia.Controls
+{
+    /// <summary>
+    /// A button control that functions as a navigateable hyperlink.
+    /// </summary>
+    [PseudoClasses(pcVisited)]
+    public class HyperlinkButton : Button
+    {
+        // See: https://www.w3schools.com/cssref/sel_visited.php
+        private const string pcVisited = ":visited";
+
+        /// <summary>
+        /// Defines the <see cref="IsVisited"/> property.
+        /// </summary>
+        public static readonly StyledProperty<bool> IsVisitedProperty =
+            AvaloniaProperty.Register<HyperlinkButton, bool>(
+                nameof(IsVisited),
+                defaultValue: false);
+
+        /// <summary>
+        /// Defines the <see cref="IsVisitedOnClick"/> property.
+        /// </summary>
+        public static readonly StyledProperty<bool> IsVisitedOnClickProperty =
+            AvaloniaProperty.Register<HyperlinkButton, bool>(
+                nameof(IsVisitedOnClick),
+                defaultValue: true);
+
+        /// <summary>
+        /// Defines the <see cref="NavigateUri"/> property.
+        /// </summary>
+        public static readonly StyledProperty<Uri?> NavigateUriProperty =
+            AvaloniaProperty.Register<HyperlinkButton, Uri?>(
+                nameof(NavigateUri),
+                defaultValue: null);
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HyperlinkButton"/> class.
+        /// </summary>
+        public HyperlinkButton()
+        {
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the <see cref="NavigateUri"/> has been visited.
+        /// </summary>
+        public bool IsVisited
```

Comment:
Open question if this should be `IsUriVisited` to be more clear.

Reply:
Probably keeping it short and matching the pseudoclass (as it does currently) is best